packages/react-core/src/components/Select/Select.tsx

```diff
@@ -1,4 +1,5 @@
 import * as React from 'react';
+import isDeepEqual from 'fast-deep-equal/react';
```

Comment:
True, it is not explicit, but it is a dependency of eslint it appears. Either way, I can use Lodash's isEqual instead, where the performance is comparable.

Reply:
I think I've done something akin to that in the memoization call in TreeView - just checking each object property which could be updated in the existing block or made into a more generic function and stored in our `util`. Lodash is also not one of our dependencies.

Reply:
@kmcfaul I saw lodash was a dependency for react-table/charts/topology, so that's where that idea came from. 
@nicolethoen informed me though that we are looking to remove lodash as a dependency across the board.

So considering this, I'll just revert to the condition that was previously there and expand upon it. I think we eventually want a more universal solution though that is agnostic to object key names.